.github/actions/commit_and_push/action.yml

```diff
@@ -0,0 +1,37 @@
+name: Commit and Push Changes
+description: 'Configures git, commits changes and pushes to a new branch'
+inputs:
+    branch-name:
+        description: 'Name of the branch to create'
+        required: true
+    commit-message:
+        description: 'Commit message'
+        required: true
+    gpg_signing_key:
+        description: 'The GPG signing key to use for signing commits'
+        required: true
+runs:
+    using: composite
+    steps:
+        - uses: ./.github/actions/configure_git
+          with:
+              gpg_signing_key: ${{ inputs.gpg_signing_key }}
+
+        - name: Commit and push changes
+          shell: bash
+          run: |
+              # Create and switch to new branch
+              git switch -c "${{ inputs.branch-name }}"
+
+              # Show status of working directory
+              echo "Current git status:"
+              git status
+
+              # Add and commit changes if there are any
+              git add --all
+              if ! git diff --cached --quiet; then
```

Comment:
Let's get rid of the if statement locura

Reply:
done